### PR TITLE
Appt 72 nextjs router

### PR DESCRIPTION
**Pull request by joefarina**
https://nhsd-jira.digital.nhs.uk/browse/APPT-72

### CODE REVIEW

File: src/client/src/Components/ServiceSelector.tsx
Comment (vicr1) [inline src/client/src/Components/ServiceSelector.tsx:1]: Doesn't appear related to title of change

File: src/client/src/Components/UserNotification.tsx
Comment (vicr1) [inline src/client/src/Components/UserNotification.tsx:1]: Unrelated change, make sure information is added to ticket otherwise these changes will not be tested

File: src/client/src/Views/TemplateListView.tsx [hunk 1]
```diff
@@ -72,7 +72,7 @@ export const TemplateListView = () => {
                   <td>{t.name}</td>
                   <td>{getOpeningDayDetails(t)}</td>
                   <td>
-                    <Link to={`edit/${t.id}`}>
+                    <Link href={`templates/edit/${t.id}`}>
```
Comment (vicr1) [inline src/client/src/Views/TemplateListView.tsx:75]: Is this change of path intentional. Assume it is because previous routes worked with relative as opposed to absolute

File: src/client/src/index.css
Comment (vicr1) [inline src/client/src/index.css:1]: This change looks like it has the potential to affect styling across the entire site. That will increase the scope of testing for the ticket

File: src/client/src/pages/_app.tsx
Comment (vicr1) [inline src/client/src/pages/_app.tsx:1]: Is there a specific reason for having an underscore in the file name?

File: src/client/src/pages/site.tsx
Comment (vicr1) [inline src/client/src/pages/site.tsx:1]: This seems somewhat confusing. The filename, component name and the underlying component being rendered don't quite agree on name. I was also wondering why we need these additional components (page components)

File: src/client/src/pages/templates/edit/[templateId].tsx
Comment (vicr1) [inline src/client/src/pages/templates/edit/[templateId].tsx:1]: These additional page component seem redundant

File: src/client/src/Components/ServiceSelector.tsx
Comment (joefarina) [reply src/client/src/Components/ServiceSelector.tsx:1]: Next.js wont allow the importing of global css in components that isn't from node modules, it threw errors in the console so as part of this work I had to change from using global css to css modules for locally scoped component styles

File: src/client/src/pages/_app.tsx
Comment (joefarina) [reply src/client/src/pages/_app.tsx:1]: Yes, its next.js convention, it overrides their default App component: https://nextjs.org/docs/pages/building-your-application/routing/custom-app

File: src/client/src/pages/templates/edit/[templateId].tsx
Comment (joefarina) [reply src/client/src/pages/templates/edit/[templateId].tsx:1]: This file allows the dynamic routing to work (Pages Router only), as nextjs's router is file system based

File: src/client/src/index.css
Comment (joefarina) [reply src/client/src/index.css:1]: We were importing multiple levels of global css, the main index.css that only contained 2 CRA derived rules, then the App.css which contained the imported NHS frontend styles and our own overrides, then a couple of components with their own stylesheets. This was how CRA setup styles, now with nextjs its cleaner to have all the global css is in one place, NHS frontend styles are imported separately and explicitly and css modules handle locally scoped imports.

The order has changed slightly (the 2 former index.css rules now apply after the NHS frontend as opposed to before) and should produce no visual changes as long as index.css is imported _after_ the NHS frontend styles in _app.tsx.

Worth noting though that the original index.css styles do impact the font-family rule and aren't part of the NHS frontend, I'd say we remove them as they are likely just vestigial from CRA and removes the issue of global css coming from multiple files

File: src/client/src/Views/TemplateListView.tsx (hunk 1 above)
Comment (joefarina) [reply src/client/src/Views/TemplateListView.tsx:75]: It is and your assumption is correct, although I anticipated nextjs to be able to handle relative paths, it doesnt work without "templates/". Adding a leading slash doesnt seem to work, this may be because both /edit and edit/[:id] route to the same component though other sources indicate relative paths arent  supported: https://stackoverflow.com/questions/69084689/relative-routing-in-next-js

**Review by vicr1 — APPROVED** [review]
